flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/typeutils/RowDataSerializer.java

```diff
@@ -314,7 +314,8 @@ public void readSnapshot(int readVersion, DataInputView in, ClassLoader userCode
             types = new LogicalType[length];
             for (int i = 0; i < length; i++) {
                 try {
-                    types[i] = InstantiationUtil.deserializeObject(stream, userCodeClassLoader);
+                    types[i] =
+                            InstantiationUtil.deserializeObject(stream, userCodeClassLoader, true);
```

Comment:
Is RowDataSerializer enough? Wouldn't it be safer to do it for all constructed types? So MapDataSerializer and ArrayDataSerializer?

Reply:
Very good point!